FunctionSignatureMap.php

```diff
@@ -2812,26 +2814,26 @@
 'Exception::__clone' => ['void'],
 'Exception::__construct' => ['void', 'message='=>'string', 'code='=>'int', 'previous='=>'?Throwable|?Exception'],
 'Exception::__toString' => ['string'],
-'Exception::getCode' => ['int'],
+'Exception::getCode' => ['int|string'],
 'Exception::getFile' => ['string'],
 'Exception::getLine' => ['int'],
 'Exception::getMessage' => ['string'],
 'Exception::getPrevious' => ['?Throwable|?Exception'],
 'Exception::getTrace' => ['array'],
 'Exception::getTraceAsString' => ['string'],
 'exec' => ['string', 'command'=>'string', '&w_output='=>'array', '&w_return_value='=>'int'],
-'exif_imagetype' => ['int', 'imagefile'=>'string'],
+'exif_imagetype' => ['int|false', 'imagefile'=>'string'],
 'exif_read_data' => ['array|false', 'filename'=>'string', 'sections_needed='=>'string', 'sub_arrays='=>'bool', 'read_thumbnail='=>'bool'],
 'exif_tagname' => ['string', 'index'=>'int'],
 'exif_thumbnail' => ['string', 'filename'=>'string', '&w_width='=>'int', '&w_height='=>'int', '&w_imagetype='=>'int'],
 'exit' => ['', 'status'=>'string|int'],
 'exp' => ['float', 'number'=>'float'],
 'expect_expectl' => ['int', 'expect'=>'resource', 'cases'=>'array', 'match='=>'array'],
 'expect_popen' => ['resource', 'command'=>'string'],
-'explode' => ['array<int,string>', 'separator'=>'string', 'str'=>'string', 'limit='=>'int'],
+'explode' => ['array<int,string>|false', 'separator'=>'string', 'str'=>'string', 'limit='=>'int'],
```

Comment:
In Phan's version of the signature map, this only contains return types that would happen if the input types were valid.
Explode would always return an array given two strings (I think)

- So changes such as `imap` and `exif` make sense (they depend on the filesystem/network and file contents and can fail), but this doesn't for Phan

This is so that `--strict-types` would correctly analyze uses of the return types. (It'd warn about param types with the right settings)

Reply:
> Explode would always return an array given two strings (I think)

Nah, not quite:

> If delimiter is an empty string (""), explode() will return FALSE.

Reply:
Should really create a plugin to check for that LiteralStringType as an argument to explode (and any other surprising edge cases in other internal functions)

Reply:
Yeah, I’ve done that in Psalm